### PR TITLE
[MRG] Added catboost example

### PR DESCRIPTION
**Pull request by scouvreur**
This PR references issue #352.

This is still a work in progress, the idea was to add an initial stub. There are still more available parameters to tune from the [Catboost docs](https://catboost.ai/docs/concepts/python-reference_parameters-list.html).

Do you have any code style requirements ?

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
> Do you have any code style requirements ?

We have [the contribution guide](https://github.com/pfnet/optuna/blob/master/CONTRIBUTING.md), so please take a look at it.

While the current official style checker is `autopep8`, we also use `yapf` in Optuna dev team. If you apply it to your code, it would be a great help.

In addition to that, we do not use type hinting for novice Python users in case of examples of Optuna. So, you do not need to add type hints for catboost examples.

If you have further questions, please let us know.

**Comment by scouvreur** [comment]
Thanks for the tips @toshihikoyanase ! That should be all solved for you now.

Would you like me to add more parameters to tune from the [Catboost docs](https://catboost.ai/docs/concepts/python-reference_parameters-list.html), or do you think this is sufficient ?

File: examples/catboost_simple.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    param = {
+        'objective': trial.suggest_categorical('objective', ['Logloss', 'CrossEntropy']),
+        'iterations': trial.suggest_int('num_leaves', 500, 2000),
+        'colsample_bylevel': trial.suggest_uniform('colsample_bylevel', 0.01, 0.1),
+        'depth': trial.suggest_int('depth', 1, 16),
+        'boosting_type': trial.suggest_categorical('boosting_type', ['Ordered', 'Plain']),
+        'learning_rate': trial.suggest_loguniform('learning_rate', 1e-6, 1.0)
+    }
+
```
Comment (sile) [inline examples/catboost_simple.py]: > Would you like me to add more parameters to tune from the Catboost docs, or do you think this is sufficient ?

IMO, it's better to add suggestions of conditional parameters to show the define-by-run feature of Optuna.
For example,
```python
bootstrap_type = trial.suggest_categorical('bootstrap_type', ['Bayesian', 'Bernoulli', 'MVS'])
params['bootstrap_type'] = bootstrap_type

if bootstrap_type == 'Bayesian':
    params['bagging_temperature'] = trial.suggest_uniform('bagging_temperature', 0, 10)
elif bootstrap_type == 'Bernoulli':
   params['subsample'] = trial.suggest_uniform('subsample', 0, 1)
elif bootstrap_type == 'MVS':
   params['mvs_head_fraction'] = trial.suggest_uniform('mvs_head_fraction', 0, 1)
```

**[NOTE]** I'm unfamiliar with Catboost, so I don't know the details of the above parameters.

File: examples/catboost_simple.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import optuna
+
+
+def objective(trial):
+    data, target = sklearn.datasets.load_breast_cancer(return_X_y=True)
+    train_x, test_x, train_y, test_y = train_test_split(data, target, test_size=0.3)
+
+    param = {
+        'objective': trial.suggest_categorical('objective', ['Logloss', 'CrossEntropy']),
+        'iterations': trial.suggest_int('num_leaves', 500, 2000),
```
Comment (sile) [inline examples/catboost_simple.py]: `num_leaves` seems a typo.
```suggestion
        'iterations': trial.suggest_int('iterations', 500, 2000),
```

Comment (sile) [reply examples/catboost_simple.py]: In addition, I think that tuning of `iterations` parameters can be eliminated.
Usually, IMO, the combination of 'setting sufficiently large (fixed) `iterations` value' and 'enabling early stopping' archives a good result with less time compared to tuning `iterations` parameter.

File: examples/catboost_simple.py (hunk 1 above)
Comment (scouvreur) [reply examples/catboost_simple.py]: Thanks for the feedback @sile, the `mvs_head_fraction` is giving me some issues although I am using the most recent version of catboost (v0.16.4). Should I remove it from the example ?

File: setup.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def get_extras_require():
         'checking': ['autopep8', 'hacking'],
         'testing': [
             'pytest', 'mock', 'bokeh', 'plotly', 'chainer>=5.0.0', 'xgboost', 'mpi4py', 'lightgbm',
-            'keras', 'mxnet', 'scikit-optimize', 'tensorflow', 'cma'
+            'catboost', 'keras', 'mxnet', 'scikit-optimize', 'tensorflow', 'cma'
```
Comment (sile) [inline setup.py]: There are no unit tests that depend on on `catboost`, so we don't have to add it to the list at this time.
```suggestion
            'keras', 'mxnet', 'scikit-optimize', 'tensorflow', 'cma'
```

File: examples/catboost_simple.py (hunk 1 above)
Comment (scouvreur) [reply examples/catboost_simple.py]: Sure ! Just removed it

**Comment by g-votte** [comment]
Merged the PR.

@scouvreur 
Thanks for your contribution!

**Pull request merged**